### PR TITLE
Update Microsoft.NETFramework.ReferenceAssemblies to latest version

### PR DESCRIPTION
**Pull request by weshaggard**
Recent agent updated version of VS which might have broken some targeting packs so testing to see if updating to the latest version helps.

cc @heaths

### CODE REVIEW

**Review by heaths — APPROVED** [review]

File: eng/Directory.Build.Common.props [hunk 1]
```diff
@@ -50,6 +50,13 @@
       CA2000; <!-- Call dispose on IDisposable objects -->
       CA2012; <!-- ValueTask should only be awaited once - conflicts with EnsureCompleted check -->
     </NoWarn>
+    <!--
+
```
Comment (heaths) [inline eng/Directory.Build.Common.props:54]: Looks like you were going to comment as to "why".

**Review by heaths — APPROVED** [review]
Glad to see it worked, though I wonder why it didn't for mine: #23380

File: eng/Directory.Build.Common.props (hunk 1 above)
Comment (weshaggard) [reply eng/Directory.Build.Common.props]: Yes I got interrupted while making the edit on github.

**Comment by weshaggard** [comment]
>Glad to see it worked, though I wonder why it didn't for mine: #23380

Looks like yours was scoped to only samples https://github.com/Azure/azure-sdk-for-net/commit/8178a5bd41f636f89dcbb811b49b1ac0da927297

Thanks @heaths for merging this but I still want to investigate whether or not it is safe to ignore MSB3245 in the general case.

**Pull request merged**